set_max_players

caillef opened the pull request.

Commit:
set_max_players + wip set_username

Commit:
Start without max players

Cheelax commented on src/systems/host.cairo:
```diff
@@ -52,44 +54,70 @@ mod host {
     impl Host of IHost<ContractState> {
-        fn create(
-            self: @ContractState, world: IWorldDispatcher, player_count: u8, player_name: felt252,
-        ) -> u32 {
```
Est ce qu'ici on pourrait envisager de laisser le player_name en optionel?
ça nous permettrait pour l'instant de laisser le player_name à entrer dans le front tant qu'on a pas terminé de le gérer différement

Cheelax commented on src/systems/host.cairo:
```diff
@@ -52,44 +54,70 @@ mod host {
+        ) { //TODO: find a way to get player from address without game_id
+        }
+
+        fn join(self: @ContractState, world: IWorldDispatcher, game_id: u32) {
```
meme question pour le player name optionel

Cheelax committed:
✨ feat: Add player_name parameter to create and join functions in IHost trait

This commit adds a new `player_name` parameter to the `create` and `join` functions in the `IHost` trait. Previously, these functions only accepted the `world` and `game_id` parameters. With this change, the functions now also accept a `player_name` parameter of type `felt252`. The `create` function creates a new host instance with the provided `world`, while the `join` function allows a player to join a specific game identified by `game_id`. The `player_name` parameter is used to set the name of the player participating in the game.

Cheelax committed:
format

Commit:
fix tests 1/2

Commit:
remove set username

Commit:
set slots to 0 when starting

bal7hazar commented on src/systems/host.cairo:
```diff
@@ -84,12 +82,36 @@ mod host {
+
+            let mut addresses = array![];
+            let mut players = store.players(game);
+            loop {
```
This loop is done 3 times at the end:
- In the store
- In this function
- In the`Game::set_max_players` function

This is very inefficient and players will pay this in fee.
We can either:
- Remove the `set_max_players` and consider only having a hard coded max player with the ability to start an unfilled game
- Skip this loop and just provide a `Span<Player>` to `Game::set_max_players`

bal7hazar committed:
✅ Revert on max players and fix tests

bal7hazar committed:
✏️ Typo
